src/components/content/catalog/services/re-register/reRegisterService.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { CloseCircleOutlined } from '@ant-design/icons';
+import { Button, Popconfirm } from 'antd';
+import catalogStyles from '../../../../../styles/catalog.module.css';
+import { ServiceTemplateDetailVo } from '../../../../../xpanse-api/generated';
+import { useReRegisterRequest } from './reRegisterMutation';
+
+function ReRegisterService({
+    id,
+    setIsViewDisabled,
+    serviceRegistrationState,
+}: {
+    id: string;
+    setIsViewDisabled: (isViewDisabled: boolean) => void;
+    serviceRegistrationState: ServiceTemplateDetailVo.serviceRegistrationState;
+}): React.JSX.Element {
+    const reRegisterRequest = useReRegisterRequest(id);
+    const reRegister = () => {
+        setIsViewDisabled(true);
+        reRegisterRequest.mutate();
+    };
+
+    return (
+        <div className={catalogStyles.updateUnregisterBtnClass}>
+            <Popconfirm
+                title='re-register the service'
+                description='Are you sure to re-register this service?'
+                cancelText='Yes'
+                okText='No'
+                onCancel={() => {
+                    reRegister();
+                }}
+            >
+                <Button
+                    icon={<CloseCircleOutlined />}
+                    type='primary'
+                    className={catalogStyles.catalogManageBtnClass}
+                    disabled={
+                        serviceRegistrationState !== ServiceTemplateDetailVo.serviceRegistrationState.UNREGISTERED
```

Comment:
in the screenshots, i see that the button is enabled even after the re-register request is submitted. The button must be disabled, if the re-register is submitted.